src/components/calcite-dropdown/calcite-dropdown.e2e.ts

```diff
@@ -689,22 +694,27 @@ describe("calcite-dropdown", () => {
       const trigger = await element.find("calcite-action[slot='dropdown-trigger'] >>> button");
       const dropdownWrapper = await page.find(`calcite-dropdown >>> .calcite-dropdown-wrapper`);
       const calciteDropdownOpen = await element.spyOnEvent("calciteDropdownOpen");
+      const calciteDropdownClose = await element.spyOnEvent("calciteDropdownClose");
 
       expect(await dropdownWrapper.isVisible()).toBe(false);
       await trigger.click();
       await page.waitForChanges();
       expect(await dropdownWrapper.isVisible()).toBe(true);
+      await page.waitForChanges();
+      expect(calciteDropdownOpen).toHaveReceivedEventTimes(1);
 
       await trigger.focus();
       await page.keyboard.press("Space");
       await page.waitForChanges();
       expect(await dropdownWrapper.isVisible()).toBe(false);
-      expect(calciteDropdownOpen).toHaveReceivedEventTimes(1);
+      await page.waitForChanges();
+      expect(calciteDropdownClose).toHaveReceivedEventTimes(1);
 
       await page.keyboard.press("Enter");
       await page.waitForChanges();
       expect(await dropdownWrapper.isVisible()).toBe(true);
-      expect(calciteDropdownOpen).toHaveReceivedEventTimes(1);
```

Comment:
Was this test failing consistently before or was it random? If it's the latter, you may need to use `page.waitForEvent` before doing assertions.

Reply:
I think it was consistent, is that right @Elijbet? Might be good to have in regardless.

Reply:
I am not sure it was consistent. It was consistently receiving one event locally and passed the tests when I pushed it up. But after that I started seeing 2 events received instead of one.